### ISSUE TITLE
Check for artithmetic base conversion bashism

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1547]
#### For bugs
- Rule Id (if any, e.g. SC1000): not yet?
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
x=$((16#"${1}"))
echo "${x}"
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Base conversion is a bashism and not portable
Suggest printf "%x" "${1}" for base 16; printf "%d" "${1}" for decimal or bc for other conversions.

**Comment by koalaman** [comment]
Nice suggestion, thanks!
